firmware/trezor.c

```diff
@@ -36,6 +38,38 @@ void __attribute__((noreturn)) __stack_chk_fail(void)
 	for (;;) {} // loop forever
 }
 
+void lock_device(void)
+{
+	static bool lockButtonDown = false;
+
+	buttonUpdate();
+	if (button.YesDown >= 200000) {
+		/* Button held down for long enough, wait for button release */
+		lockButtonDown = true;
+	} else if (lockButtonDown && button.YesUp) {
+		/* Button released */
+		lockButtonDown = false;
+
+		layoutDialog(&bmp_icon_question, "Cancel", "Lock Device", NULL, "Do you really want to", "lock your TREZOR?", NULL, NULL, NULL, NULL);
```

Comment:
I must admit I didn't try it yet :-). However generally we want to avoid unnecessary button confirmations, just to teachn people that pressing button is something they should care about.